Hint to use `addEventListener` instead of the onchange property (#28105)
mdn/content: files/en-us/web/api/screenorientation/change_event/index.md

---
title: "ScreenOrientation: change event"
short-title: change
slug: Web/API/ScreenOrientation/change_event
page-type: web-api-event
browser-compat: api.ScreenOrientation.change_event
---

> {{securecontext_header}}{{APIRef("Screen Orientation API")}}{{SeeCompatTable}}

Reviewer: ```suggestion
{{APIRef("Screen Orientation API")}}
```

(this will get rid of those banners you mentioned in https://github.com/mdn/content/pull/28105#issuecomment-1648551720 )

> The **`change`** event of the {{domxref("ScreenOrientation")}} fires when the orientation of the screen has changed, for example when a user rotates his mobile phone.

Reviewer: ```suggestion
The **`change`** event of the {{domxref("ScreenOrientation")}} interface fires when the orientation of the screen has changed, for example when a user rotates their mobile phone.
```

## Syntax

Use the event name in methods like {{domxref("EventTarget.addEventListener", "addEventListener()")}}, or set an event handler property.

```js
addEventListener("change", (event) => {});

onchange = (event) => {};
```

## Event type

A generic {{domxref("Event")}}.

## Example

> In the following example, the `change` callback prints the new {{DOMxRef("ScreenOrientation.type", "screen orientation type", "", "nocode")}} and {{DOMxRef("ScreenOrientation.angle", "angle", "", "nocode")}}

Reviewer: ```suggestion
In the following example, the `change` callback prints the new {{DOMxRef("ScreenOrientation.type", "screen orientation type", "", "nocode")}} and {{DOMxRef("ScreenOrientation.angle", "angle", "", "nocode")}}.
```

> In the following example, the `change` callback prints the new {{DOMxRef("ScreenOrientation.type", "screen orientation type")}} and {{DOMxRef("ScreenOrientation.angle", "angle")}}

Reviewer: ```suggestion
In the following example, the `change` callback prints the new {{DOMxRef("ScreenOrientation.type", "screen orientation type", "", "nocode")}} and {{DOMxRef("ScreenOrientation.angle", "angle", "", "nocode")}}
```

(since these are not identifier names, we should not code-format them.)
